src/components/metrics/card/name/index.js

```diff
@@ -1,108 +1,117 @@
-  }
-
-  _editable() {
-    if (this.state.persistEditing) {
-      return this.state.editing
-    } else {
-      return this.props.editable && this.state.editing
-    }
+    value: this.props.name
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.name === this.state.value) {this.setState({value: nextProps.name})}
+    if (!nextProps.inSelectedCell && (this.props.name !== nextProps.name) && (nextProps.name !== this.value())) {
+      this.refs.SimpleEditor.changePlainText(nextProps.name)
+    }
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
-    return ((nextProps.name !== this.props.name) ||
-            (nextProps.inSelectedCell !== this.props.inSelectedCell) ||
-            (nextProps.editable !== this.props.editable) ||
-            (nextState.value !== this.state.value) ||
-            (nextState.editing !== this.state.editing))
+  componentWillUnmount() {
+    this.handleSubmit()
   }
 
   handleSubmit() {
     if (this._hasChanged()){
-      this.props.onChange({name: this.state.value})
+      this.props.onChange({name: this.value()})
     }
-    this.setState({editing: false, persistEditing: false})
   }
 
   _hasChanged() {
-    return (this.state.value != this.props.name)
-  }
-
-  _handleMouseLeave() {
-    if (!(this.state.persistEditing || this._hasChanged())) {
-      this.setState({editing: false})
-    }
+    return (this.value() != this.props.name)
   }
 
   hasContent() {
-    return !_.isEmpty(this.state.value)
-  }
-
-  componentWillUnmount() {
-    this.handleSubmit()
+    return !_.isEmpty(this.value())
   }
 
-  onChange(e) {
-    this.setState({value: e.target.value})
+  value() {
+    return this.refs.SimpleEditor.getPlainText()
   }
 
   handleKeyDown(e) {
+    e.stopPropagation()
+    this.props.heightHasChanged()
     const ENTER = (e) => ((e.keyCode === 13) && !e.shiftKey)
     if (ENTER(e)){
       e.stopPropagation()
       this.props.jumpSection()
     }
   }
 
-
   render() {
+    const isClickable = !this.props.anotherFunctionSelected
     return (
-      <div className='MetricName'>
-        {this._editable() &&
-          <TextArea
-            onBlur={this.handleSubmit.bind(this)}
-            onChange={this.onChange.bind(this)}
-            onKeyDown={this.handleKeyDown.bind(this)}
-            onMouseOut={this._handleMouseLeave.bind(this)}
-            onMouseDown={() => {this.setState({persistEditing: true})}}
-            placeholder={'name'}
-            ref={'input'}
-            tabIndex={2}
-            value={this.state.value}
-          />
-        }
-        {!this._editable() &&
-          <div className={`static${!this.state.value ? ' default-value' : ''}`}
-            onMouseOver={() => {if (!this.state.editing) {this.setState({editing: true})}}}n
-            onMouseDown={() => {this.setState({persistEditing: true})}}
-          >
-            {(this.state.value || 'name').replace(/ /g, "\u2005")}
-          </div>
-        }
-      </div>
+      <span
```

Comment:
No strong reason, but I generally prefer spans for things that won't take extra space.  They're less flexible, which makes that more explicit imo.